settings: Make list header sections consistent.

showell opened the pull request:
This affects any settings panel (personal or org) that has some kind of list of items, so the best way to QA it is to log in as Iago and hit every panel.

alya commented:
On the emoji panel, we should be consistent on whether we use "emoji" or "emojis" as the plural of "emoji". Either one is [accepted](https://www.merriam-webster.com/dictionary/emoji), but I think I prefer "emoji" in our context. So let's change "Custom emojis" and "Filter emojis" accordingly.

timabbott commented on static/templates/settings/data_exports_admin.hbs:
```diff
@@ -1,5 +1,5 @@
 <div id="data-exports" class="settings-section" data-name="data-exports-admin">
-    <h3>{{t "Data exports" }}
+    <h3>{{t "Export data" }}
```
We may want this to be "Export organization data" or something similar.

showell replied:
Fixing.  I'm not sure what the best name is here, but I wanted something other than "Data exports" since I introduced it as the title of the list.  I tried not to go down the rabbit hole of fixing every wording issue that I encountered in the settings panels.  The wording right below that title is kind of awkward, but that's a problem for another day.

alya commented:
How about we make the placeholder text in the filter always match the table heading?

* Default streams: Filter streams -> Filter default streams
* Uploaded files: Filter uploads -> Filter uploaded files

alya commented:
Other than that, looks good to me!

timabbott commented on static/styles/settings.css:
```diff
@@ -2040,3 +2029,21 @@ input[type="checkbox"] {
+        font-size: 1.3em;
+        line-height: 1.3;
+        margin-top: 6px;
+    }
```
What's the thinking of using sizing CSS here vs. the common CSS we'd get by preserving the previous `h3` tags? We have some CSS related to h3s in settings, which possibly can be deleted if we prefer this approach.

showell replied:
The main problem here is that we're trying to make sure that the title on the left mostly vertically aligns with filter box on the right, as well as having roughly the same dimensions.  You have to do fiddly stuff like the `margin-top` to make it look nice, and I suspect a designer could do an iteration on this to make it look nicer.

I really don't want to couple this problem to all the other h3 tags in our system.

timabbott replied:
We basically only have H3 tags in settings. The main reason I mention this is that we have special CSS for e.g. `help_settings_link` for H3 tags in settings pages.

timabbott replied:
Yeah, and this is in fact a bug -- the "Deactivated users" heading has a blue link because that CSS is no longer applying.

I think it'd be better to fix by preserving the h3s; that's semantically the role of these elements; almost all of our H3s in Zulip are in the settings system, the CSS is scoped to settings pages, and these H3s are primarily for the exact section heading purpose we have here.

showell replied:
I went to H4 tags because H3 tags look too large to me.

timabbott commented on static/templates/settings/data_exports_admin.hbs:
```diff
@@ -29,8 +29,13 @@
+
+    <div class="settings_panel_list_header">
+        <div class="list_title">{{t "Data exports"}}</div>
+        <input type="hidden" class="search" placeholder="{{t 'Filter exports' }}"
```
This is preexisting, but `type="hidden"` seems unlikely to be correct.

showell replied:
Yeah, this is basically the only place where we hide the filter button until there are actual items in the list.  This is outside the scope of my changes, as I don't want to go down the rabbit hole of mucking with the JS here, and this is a pretty rarely used screen.  In fairness to the decision here it's quite rare to need a filter for data exports, so it's possible that the best decision is to just eliminate the filter altogether.  That's outside the scope of a code sweep.

timabbott replied:
OK, sounds good. Eventually I think we'll want to just rework this to be a formal component that works more like `settings_checkbox`, where you just pass in a couple options, and features like that hiding detail can be part of the common system. Agree it isn't something we should change here, as long as it's not obviously a bug.

timabbott commented on static/templates/settings/profile_field_settings_admin.hbs:
```diff
@@ -1,6 +1,8 @@
-    <h3 class="inline-block">{{t "Custom profile fields" }}</h3>
     <div class="alert-notification" id="admin-profile-field-status"></div>
+    <div class="settings_panel_list_header">
+        <div class="list_title">{{t "Custom profile fields"}}</div>
```
Should we call this class `settings_panel_list_title` just for consistency with the `settings_panel_list_header` class?

showell replied:
I value brevity here.  I gave the container a long name to ensure it always stays unique.  The `list_title` class is also unique, and it's consistent with the short class for the search widget ("search") being short.  I understand that `list_title` may not always remain unique, but it's nested in the CSS selectors.

timabbott commented:
Both of Alya's wording suggestions sound good to me.

showell commented:
> How about we make the placeholder text in the filter always match the table heading?
> 
>     * Default streams: Filter streams -> Filter default streams
> 
>     * Uploaded files: Filter uploads -> Filter uploaded files

Fixing.

showell commented:
This is ready for review again. I re-worded the two filters, changed the "emojis" spelling, and changed "Export data" to "Export organization".  For all the other comments, I explained my reasoning above.

alya commented:
Cool, looks good to me.

showell commented:
I changed the `<div>` tags to `<h4>` tags.  The `<h4>` tags are a tad too big, and using `<h3>` tags here would just be ugly.

I think I changed the correct CSS rule for "Deactivated Users" to be less picky about h3 vs. h4, but please double check that in particular.

timabbott commented:
@alya I think we need your input on the H3 vs H4 question. Here's what this now looks like:

![image](https://user-images.githubusercontent.com/2746074/156841231-a5974b56-beb4-48b1-8b77-6318e49de420.png)

My view is these two headings should be the same size; we already have multiple H3s (aka the larger style) on various settings pages by design, and using a smaller size for the table headings both looks off to me and also suggests that it's inside the "Add an X" heading, which I think isn't what we want.

alya commented:
Yeah, I think we should stick with the H3s, at least for now.

Do we use the H4s anywhere besides the Create stream menu?

alya commented:
I'd be open to using another heading style for the table, I guess, but I also just don't think the currently H4 styling looks very good.

showell commented:
Ok, back to H3 headings.  Ready for another round of review.

timabbott commented on static/styles/settings.css:
```diff
@@ -2040,3 +2029,18 @@ input[type="checkbox"] {
+    input.search {
+        float: right;
+        font-size: 1em;
+        max-width: 160px;
```
I changed this 160px before merging because I'd done the same in the subscriptions PR (where you'd done 206px=>130px, and I decided that was a bit too small; forgot to note it), and we'll hopefully eventually deduplicate CSS between the two.

timabbott commented:
Great, merged after the tiny change noted above.  Thanks @showell!
